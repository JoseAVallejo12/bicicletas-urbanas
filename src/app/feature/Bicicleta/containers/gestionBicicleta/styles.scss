.bicisGrid {
  flex-grow: 1;
}

.MuiGrid-grid-xs-3 {
  flex-grow: 0;
  max-width: none !important;
  flex-basis: auto !important;
}