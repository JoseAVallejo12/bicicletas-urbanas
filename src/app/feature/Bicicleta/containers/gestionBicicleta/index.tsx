import * as PropTypes from 'prop-types';
import * as React from 'react';
import { Bicicleta } from '../../models/Bicicleta';
import { useEffect } from 'react';

import { MediaCardBici } from 'app/shared/components/Card/index';
import './styles.scss'
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 3,
    },
    paper: {
      textAlign: 'center',
      color: theme.palette.text.secondary,
    },
  }),
);

interface GestionBicicletaProps {
  bicicleta: Array<Bicicleta>;
  listarBici: () => void;
}

export const GestionBicicleta: React.FC<GestionBicicletaProps> = ({ bicicleta, listarBici }) => {

  const classes = useStyles();
  useEffect(() => {
    listarBici();
  }, [listarBici]);

  return (
    <Grid container className={classes.root} spacing={3}>
      <Grid container justify="center" spacing={3}>
        {(bicicleta.length > 0) ? (bicicleta.map((bici) => (
          <Grid item xs={3} key={bici.id}>
            <Paper className={classes.paper} key={bici.id}>
              <MediaCardBici key={bici.id} bici={bici} listarBici={listarBici} />
            </Paper>
          </Grid>
        ))) : (undefined)}
      </Grid>
    </Grid>
  );
};

GestionBicicleta.propTypes = {
  bicicleta: PropTypes.array.isRequired,
  listarBici: PropTypes.func.isRequired
};
