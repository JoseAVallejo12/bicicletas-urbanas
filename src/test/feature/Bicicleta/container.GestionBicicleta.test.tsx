import * as React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';

import { GestionBicicleta } from '../../../app/feature/Bicicleta/containers/gestionBicicleta/index';

describe(' <GestionBicicletas />Test', () => {
  let componentWrapper: ShallowWrapper;
  const bicicleta = {
    id: 1,
    marca: 'Jeep',
    serial: 'BALTORO 29 V20',
    color: 'Roja',
    almacenActual: 1,
    fechaCompra: '2021-01-19T05:00:00.000Z',
    estado: 'alquilada',
    valorHora: '5000',
    descripcion: 'Bicicleta de Montaña Jeep Baltoro 29 pulgadas Género: Unisex Material del marco: Aluminio, Número de cambios: 21, Tipo de freno: Disco mecánico, Suspensión: Delantera'
  };

  afterEach(() => {
    componentWrapper.unmount();
  });

  it('should match snapshot', () => {
    componentWrapper = shallow(<GestionBicicleta bicicleta={[bicicleta]} listarBici={() => {}}/>)
    expect(componentWrapper).toMatchSnapshot();
  });

});

export {};
