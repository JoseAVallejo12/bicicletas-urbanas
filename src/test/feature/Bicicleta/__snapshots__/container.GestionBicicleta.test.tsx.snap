// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <GestionBicicletas />Test should match snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-root-1"
  container={true}
  spacing={3}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="center"
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="1"
      xs={3}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2"
        key="1"
      >
        <MediaCardBici
          bici={
            Object {
              "almacenActual": 1,
              "color": "Roja",
              "descripcion": "Bicicleta de Montaña Jeep Baltoro 29 pulgadas Género: Unisex Material del marco: Aluminio, Número de cambios: 21, Tipo de freno: Disco mecánico, Suspensión: Delantera",
              "estado": "alquilada",
              "fechaCompra": "2021-01-19T05:00:00.000Z",
              "id": 1,
              "marca": "Jeep",
              "serial": "BALTORO 29 V20",
              "valorHora": "5000",
            }
          }
          key="1"
          listarBici={[Function]}
        />
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
